poptorch_experimental_addons/_impl/collectives.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+import torch
+import poptorch
+
+
+def replicated_all_gather(x: torch.Tensor, replication_factor: int) -> torch.Tensor:
```

Comment:
I'd love to get rid of `replication_factor` here! I might ferret around if that's OK? But did you ask anyone/see any hope?

Reply:
Sure, really happy to take suggestions here. @awf suggested a global tensor registry, but I thought this might be addressed in a future PR if we decide this is the simplest solution.